qiskit_nature/utils/linalg.py

```diff
@@ -179,3 +179,136 @@ def _swap_columns(matrix: np.ndarray, i: int, j: int) -> None:
     column_i = matrix[:, i].copy()
     column_j = matrix[:, j].copy()
     matrix[:, i], matrix[:, j] = column_j, column_i
+
+
+def modified_cholesky(
+    mat: np.ndarray, *, error_threshold: float = 1e-8, max_vecs: int | None = None
+) -> np.ndarray:
+    r"""Modified Cholesky decomposition.
+
+    The modified Cholesky decomposition of a square matrix :math:`M` has the form
+
+    .. math::
+        M = \sum_{i=1}^N v_i v_i^\dagger
+
+    where each :math:`v_i` is a vector. :math:`M` must be positive definite.
+    No checking is performed to verify whether :math:`M` is positive definite.
```

Comment:
How will this currently appear to the user when not satisfying this? Will it outright break? If so, what will the error message be? If not, does that mean it silently "works"?

Reply:
It silently "works" but returns nonsense. This matches the behavior of e.g. `np.linalg.cholesky` so should not be surprising to users.

Reply:
I suggest adding a `.. warning::` admonition just like it is done in the other docstring to emphasize this further.

Reply:
Done.